ibis/spark/compiler.py

```diff
@@ -26,13 +68,356 @@ class SparkContext(ImpalaContext):
     pass
 
 
+def _is_inf(translator, expr):
+    arg, = expr.op().args
+    return '{} = {} or {} = {}'.format(
+        *map(
+            translator.translate,
+            [
+                arg,
+                ibis.literal(math.inf, type='double'),
+                arg,
+                ibis.literal(-math.inf, type='double'),
+            ]
+        )
+    )
+
+
+_sql_type_names = impala_compiler._sql_type_names.copy()
+
+_sql_type_names.update({
+    'date': 'date',
+})
+
+
+def _cast(translator, expr):
+    op = expr.op()
+    arg, target_type = op.args
+    arg_formatted = translator.translate(arg)
+
+    if isinstance(arg, ir.CategoryValue) and target_type == dt.int32:
+        return arg_formatted
+    if isinstance(arg, ir.TemporalValue) and target_type == dt.int64:
```

Comment:
Do we have a test that hits this branch?

Reply:
Created issue.